### ISSUE TITLE
Is it correct way import plugin as suggested in README?

### ISSUE THREAD

**Issue by andrii-lundiak** [Brugui7/cypress-log-filter#3]
TL;DR
If it's really plugin per se, then it should be imported via `plugins/index.js` (and NOT via `support/e2e.js).

Currently when I try to do it I got error:
![image](https://github.com/Brugui7/cypress-log-filter/assets/85486996/26489ca9-4f63-4811-89ab-148273bbf87b)

```
ReferenceError: before is not defined
    at C:\projects_win\end-to-end\node_modules\cypress-log-filter\dist\index.js:1:781
    at Object. (C:\projects_win\end-to-end\node_modules\cypress-log-filter\dist\index.js:1:2935)
    at Module._compile (node:internal/modules/cjs/loader:1155:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
    at Object.require.extensions. [as .js] (C:\Users\alundiak\AppData\Local\Cypress\Cache\12.17.2\Cypress\resources\app\node_modules\ts-node\dist\index.js:851:20)
    at Module.load (node:internal/modules/cjs/loader:1033:32)
    at Function.Module._load (node:internal/modules/cjs/loader:868:12)
    at Module.require (node:internal/modules/cjs/loader:1057:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object. (C:\projects_win\end-to-end\cypress\plugins\index.js:19:1)
    at Module._compile (node:internal/modules/cjs/loader:1155:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
    at Object.require.extensions. [as .js] (C:\Users\alundiak\AppData\Local\Cypress\Cache\12.17.2\Cypress\resources\app\node_modules\ts-node\dist\index.js:851:20)
    at Module.load (node:internal/modules/cjs/loader:1033:32)
    at Function.Module._load (node:internal/modules/cjs/loader:868:12)
    at Module.require (node:internal/modules/cjs/loader:1057:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at setupNodeEvents (C:\projects_win\end-to-end\cypress.config.ts:39:14)
    at C:\Users\alundiak\AppData\Local\Cypress\Cache\12.17.2\Cypress\resources\app\node_modules\@packages\server\lib\plugins\child\run_plugins.js:122:14
    at tryCatcher (C:\Users\alundiak\AppData\Local\Cypress\Cache\12.17.2\Cypress\resources\app\node_modules\bluebird\js\release\util.js:16:23)
    at Function.Promise.attempt.Promise.try (C:\Users\alundiak\AppData\Local\Cypress\Cache\12.17.2\Cypress\resources\app\node_modules\bluebird\js\release\method.js:39:29)
    at RunPlugins.load (C:\Users\alundiak\AppData\Local\Cypress\Cache\12.17.2\Cypress\resources\app\node_modules\@packages\server\lib\plugins\child\run_plugins.js:119:9)
```

**When I execute code on CI** (meaning target is `cypress run` command) I also have same error:

![image](https://github.com/Brugui7/cypress-log-filter/assets/85486996/99e94da4-931e-4f66-b9e7-1d4d527df97c)

**Yes**, when I import in `support/e2e.js` **I have no error in GUI**, but I also **DON'T have** dropdown for log level select.
![image](https://github.com/Brugui7/cypress-log-filter/assets/85486996/44a7a33e-149f-4aac-9f93-7173317934ac)

And YES, I did configure `cypress.config.ts`:

```
e2e: {
    setupNodeEvents(on, config) {
      return require('./cypress/plugins/index.js')(on, config)
    },
    env: {
      logLevel: "ERROR"
    },
    excludeSpecPattern: ['*.js'],
    specPattern: 'cypress/e2e/**/*.spec.ts',
    baseUrl: process.env.CYPRESS_BASE_URL
  },
```
**And I also** set `CYPRESS_LOG_LEVEL` as alternative in my `.env`.

I use Cypress `v12.17.2`, NodeJS v16 and v18. I execute e2e tests on Windows but also on CI/Ubuntu.

@Brugui7 what do you think about above?
I see that your plugin depends on at least `v12.5.1` so it should have been OK when you migrated to new API after v10, v11 I assume. No?
Am I missing something? Have you missed info about `plugins/index.js` in readme?

**Comment by Brugui7** [comment]
Hi @andrii-lundiak and thanks for opening this issue,

Regarding

> If it's really plugin per se, then it should be imported via plugins/index.js 

Configuring plugins via cypress/plugins/index.js is no longer supported as of Cypress version 10.0.0. [More info](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Plugins-file). Also there are lots of Cypress plugins that are imported in the `e2e.js` file.

Regarding:

>  when I import in support/e2e.js I have no error in GUI, but I also DON'T have dropdown for log level select.

The dropdown should be visible once you select a spec to run, the screenshot you shared  is from the spec selections, not an actual spec in execution.

And yes,  the plugin should work with `v12.17.2`, this screenshot is from `v12.17.1`

![image](https://github.com/Brugui7/cypress-log-filter/assets/18025825/e9e4b1f6-b5bb-4292-9ec1-038d7316e692)

Please let me know if this solves your issue.
Thanks!

**Comment by andrii-lundiak** [comment]
OK, in regards to API. I don't use much of plugins. So far only `cypress-dotenv`. Maybe that is also redundant.

And yes, I do confirm, when I finally execute dedicated spec file, the I really have that dropdown.
![image](https://github.com/Brugui7/cypress-log-filter/assets/85486996/512c58e7-f321-4068-aeec-ce411c2b5c17)


Thanks.
